File: LivePriceBackend/DTOs/CParityRule/CParityRuleUpdateModel.cs
using LivePriceBackend.Constants.Enums;

namespace LivePriceBackend.DTOs.CParityRule;

public class CParityRuleUpdateModel
{
    public bool IsVisible { get; set; }
    public SpreadRuleType? SpreadRuleType { get; set; }
    public decimal? SpreadForAsk { get; set; }
    public decimal? SpreadForBid { get; set; }
}

File: live_price_frontend/lib/modules/customer_panel/models/c_parity_rule_update_model.dart
import 'c_parity_rule_view_model.dart';

class CParityRuleUpdateModel {
  final bool isVisible;
  final SpreadRuleType? spreadRuleType;
  final double? spreadForAsk;
  final double? spreadForBid;

  CParityRuleUpdateModel({
    required this.isVisible,
    this.spreadRuleType,
    this.spreadForAsk,
    this.spreadForBid,
  });

  Map<String, dynamic> toJson() {
    return {
      'isVisible': isVisible,
      'spreadRuleType': spreadRuleType?.id,
      'spreadForAsk': spreadForAsk,
      'spreadForBid': spreadForBid,
    };
  }
}
